### PR TITLE
[DE-2864] fix trailing comma when no columns to pivot

### PR DESCRIPTION
**Pull request by jennaallen**
Fixes trailing comma if no columns to pivot.

### CODE REVIEW

**Review by emekdahl — APPROVED** [review]
@jennaallen and I reviewed this against the prior change and discussed why this change should succeed

File: macros/object_pivot.sql [hunk 1]
```diff
@@ -12,7 +12,8 @@
 
     select
     {%- for ic in include_columns %}
-        {{ ic }},
+        {{ ic }}{%- if not loop.last -%},{%- endif -%}
```
Comment (avyfain) [inline macros/object_pivot.sql:15]: should this just be a `join`? https://jinja.palletsprojects.com/en/3.0.x/templates/#jinja-filters.join

Comment (emekdahl) [reply macros/object_pivot.sql:15]: that does seem safer

Comment (jennaallen) [reply macros/object_pivot.sql:15]: Tell me more about why it is safer. We use this pattern without the join all over the place
https://docs.getdbt.com/guides/advanced/using-jinja#use-looplast-to-avoid-trailing-commas

**Comment by treeforest456** [comment]
wondering is there a way to test this? like a csv and a custom test like what we have in dbt?
not blocking but just thinking that would be a wonderful thing to have?

**Review by emekdahl — APPROVED** [review]

File: macros/object_pivot.sql (hunk 1 above)
Comment (emekdahl) [reply macros/object_pivot.sql:15]: i approved before this was suggested. i'm comfortable with the original loop pattern. i'm not sure why my approval got removed so i re-added it.

**Comment by jennaallen** [comment]
> wondering is there a way to test this? like a csv and a custom test like what we have in dbt? not blocking but just thinking that would be a wonderful thing to have?

Testing here is a bit tough. What I do now is update the branch for this package in my local environment, run dpt deps so this branch's changes come through. Then, in this case, I ran some models that use the pivot and see if they ran correctly and I also look at the compiled code. For this, I looked at the models giving us issues and ones in the FRED data where we have no issues.

**Comment by avyfain** [comment]
We also need to create a new tag here and bump this in DWH's packages

**Pull request merged**